src/Controller/LoginController.php

```diff
@@ -164,6 +164,20 @@ public function login(
         // This will be used to come back from the AuthSource login or from the Processing Chain
         $returnToUrl = $this->getReturnUrl($request, $sessionTicket);
 
+        /*
+         * CAS gateway behavior:
+         * If gateway=true, service is valid, and the user is not authenticated,
+         * redirect immediately to the service URL with NO query parameters or fragment.
+         */
+        if ($gateway === true && $serviceUrl !== null && !$this->authSource->isAuthenticated()) {
+            $cleanServiceUrl = $this->stripQueryParameters($serviceUrl);
```

Comment:
This doesn't line up with my reading of the spec. I don't think you need to manipulate the service url.  The service url is allowed to have query parameters.  I think the spec just says "do not add a ticket query parameter" and redirect the user to that serviceUrl.

From https://apereo.github.io/cas/7.3.x/protocol/CAS-Protocol-Specification.html

>  If the client does not have a single sign-on session with CAS, and a non-interactive authentication cannot be established, CAS MUST redirect the client to the URL specified by the service parameter with no “ticket” parameter appended to the URL.